homeassistant/components/yale/config_flow.py

```diff
@@ -1,282 +1,61 @@
 from typing import Any
 
-import aiohttp
-import voluptuous as vol
-from yalexs.authenticator import ValidationResult
-from yalexs.const import BRANDS
-from yalexs.manager.exceptions import CannotConnect, InvalidAuth, RequireValidation
+import jwt
 
-from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
-from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
-from homeassistant.core import callback
+from homeassistant.config_entries import ConfigEntry, ConfigFlowResult
+from homeassistant.helpers import config_entry_oauth2_flow
 
-from .const import (
-    CONF_ACCESS_TOKEN_CACHE_FILE,
-    CONF_BRAND,
-    CONF_LOGIN_METHOD,
-    DEFAULT_BRAND,
-    DEFAULT_LOGIN_METHOD,
-    DOMAIN,
-    LOGIN_METHODS,
-    VERIFICATION_CODE_KEY,
-)
-from .gateway import YaleGateway
-from .util import async_create_yale_clientsession
+from .const import DOMAIN
 
 _LOGGER = logging.getLogger(__name__)
 
 
-async def async_validate_input(
-    data: dict[str, Any], yale_gateway: YaleGateway
-) -> dict[str, Any]:
-    """Validate the user input allows us to connect.
-
-    Data has the keys from DATA_SCHEMA with values provided by the user.
-
-    Request configuration steps from the user.
-    """
-    assert yale_gateway.authenticator is not None
-    authenticator = yale_gateway.authenticator
-    if (code := data.get(VERIFICATION_CODE_KEY)) is not None:
-        result = await authenticator.async_validate_verification_code(code)
-        _LOGGER.debug("Verification code validation: %s", result)
-        if result != ValidationResult.VALIDATED:
-            raise RequireValidation
-
-    try:
-        await yale_gateway.async_authenticate()
-    except RequireValidation:
-        _LOGGER.debug(
-            "Requesting new verification code for %s via %s",
-            data.get(CONF_USERNAME),
-            data.get(CONF_LOGIN_METHOD),
-        )
-        if code is None:
-            await yale_gateway.authenticator.async_send_verification_code()
-        raise
-
-    return {
-        "title": data.get(CONF_USERNAME),
-        "data": yale_gateway.config_entry(),
-    }
-
-
-@dataclass(slots=True)
-class ValidateResult:
-    """Result from validation."""
-
-    validation_required: bool
-    info: dict[str, Any]
-    errors: dict[str, str]
-    description_placeholders: dict[str, str]
-
-
-class YaleConfigFlow(ConfigFlow, domain=DOMAIN):
+class YaleConfigFlow(config_entry_oauth2_flow.AbstractOAuth2FlowHandler, domain=DOMAIN):
     """Handle a config flow for Yale."""
 
     VERSION = 1
+    DOMAIN = DOMAIN
 
     def __init__(self) -> None:
-        """Store an YaleGateway()."""
-        self._yale_gateway: YaleGateway | None = None
-        self._aiohttp_session: aiohttp.ClientSession | None = None
-        self._user_auth_details: dict[str, Any] = {}
-        self._needs_reset = True
-        self._mode: str | None = None
+        """Instantiate config flow."""
+        self.reauth_entry: ConfigEntry | None = None
```

Comment:
Can be set outside of the constructor, after that the constructor can be removed

Reply:
70694237010